scripts/simplify_network.py

```diff
@@ -743,6 +746,182 @@ def drop_isolated_nodes(n, threshold):
+    multicountry_subnetworks = subnetw_ac_df[
+        subnetw_ac_df.index.droplevel("country").duplicated()
+    ].index.get_level_values("sub_network")
+
+    subnetw_ac_monocountry_df = subnetw_ac_df.loc[
+        subnetw_ac_df.index.get_level_values("sub_network").difference(
+            multicountry_subnetworks
+        )
+    ]
+
+    island_sbntw = []
+    for cnt in buses_df.country.unique():
+        ## TODO May be worth to investigate a threshold for the load share
+        ## load may attached to multi-country networks only
+        # country_load = (
+        #    n.loads_t.p_set[
+        #        n.loads_t.p_set.columns.intersection(
+        #            n.buses.index[n.buses.country == cnt]
+        #        )
+        #    ]
+        #    .sum()
+        #    .sum()
+        # )
+
+        sbntw = subnetw_ac_monocountry_df.query(
+            "country == @cnt"
+        ).index.get_level_values("sub_network")
+
+        i_island_ntw = [
+            s
+            for s in sbntw
+            if (
+                n.loads_t.p_set[n.buses[n.buses.sub_network == s].bus_id].mean().mean()
+                < threshold
+            )
+        ]
+        island_sbntw.extend(i_island_ntw)
+
+    i_islands = n.buses[n.buses.sub_network.isin(island_sbntw)].bus_id
+
+    return i_islands
+
+
+def transform_to_gdf(buses_df, i_buses, network_crs):
+    buses_df["bus_id"] = buses_df.index
+    points_buses = buses_df.loc[i_buses, ["bus_id", "x", "y", "country"]]
+    gdf_buses = gpd.GeoDataFrame(
+        points_buses,
+        geometry=gpd.points_from_xy(points_buses.x, points_buses.y),
+        crs=network_crs,
+    )
+    return gdf_buses
+
+
+def merge_into_network(n, threshold, aggregation_strategies=dict()):
+    """
+    Find isolated AC nodes and sub-networks in the network and merge those of
+    them which have load value and a number of buses below than the specified
+    thresholds into a backbone network.
+
+    Parameters
+    ----------
+    n : PyPSA.Network
+        Original network
+    threshold : float
+        Load power used as a threshold to merge isolated nodes
+    aggregation_strategies: dictionary
+        Functions to be applied to calculate parameters of the aggregated grid
+
+    Returns
+    -------
+    modified network
+    """
+    # keep original values of the overall load and generation in the network
+    # to track changes due to drop of buses
+    generators_mean_origin = n.generators.p_nom.mean()
+    load_mean_origin = n.loads_t.p_set.mean().mean()
+
+    network_crs = snakemake.params.geo_crs
+
+    n.determine_network_topology()
+
+    i_islands = find_isolated_sub_networks(n=n, threshold=threshold)
+
+    # return the original network if no isolated nodes are detected
+    if len(i_islands) == 0:
+        return n, n.buses.index.to_series()
+
+    gdf_islands = transform_to_gdf(
+        buses_df=n.buses, i_buses=i_islands, network_crs=network_crs
+    )
```

Comment:
The main purpose for `transform_to_gdf( )` is to keep pandas-to-geopandas transformations away from the "main" code. 

Initially, I have [tried without it](https://github.com/pypsa-meets-earth/pypsa-earth/blob/dc9be87883c95155db589aafc2a294d13b83ff23/scripts/simplify_network.py#L787-L798), but geopandas stuff looked a bit distractive `gpd.points_from_xy(points_buses.x, points_buses.y)`. So, I have wrapped-up this part into a function to focus on topology-related part in the higher-level `merge_into_network( )` function.

Using `i_buses` as an argument, has been just the next step: since we have an additional function anyway, why not to transfer also subsetting to it? :D

Does it sound reasonable for you? :)